frontends/p4/validateParsedProgram.cpp

```diff
@@ -48,6 +48,18 @@ void ValidateParsedProgram::postorder(const IR::Method* m) {
     }
 }
 
+/// Check that verify is not used outside of parsers
+void ValidateParsedProgram::postorder(const IR::MethodCallExpression* e) {
```

Comment:
This will work with the current language definition, but it would be more robust to use MethodInstance to resolve the method. This check could probably be better placed in `TypeInference::postorder(IR::MethodCallExpression* e)`. There is already a call to MethodInstance, one just needs to add a check for this particular built-in method.